rero_ils/modules/entities/remote_entities/sync.py

```diff
@@ -294,15 +277,14 @@ def sync_record(self, pid):
         :rtype: integer, boolean, x.
         """
         # close db session to prevent psycopg2.OperationalError.
-        # a new session will be opend automaticly.
+        # a new session will be open automatically.
         db.session.close()
         doc_updated = set()
         updated = error = False
         try:
-            # get contribution in db
-            entity = RemoteEntity.get_record_by_pid(pid)
-            if not entity:
-                raise Exception(f'ERROR MEF {pid} does not exists in db.')
+            if not (entity := RemoteEntity.get_record_by_pid(pid)):
+                raise RecordNotFound(RemoteEntity, entity.pid)
```

Comment:
If the entity is not found `get_record_by_pid` return 'None'
`entity.pid` raise `AttributeError`.
It Should be change to `raise RecordNotFound(RemoteEntity, pid)`

Reply:
correct ! Hawk-eye !!